nestkernel/simulation_manager.cpp

```diff
@@ -1038,7 +1038,7 @@ nest::SimulationManager::update_()
           i != kernel().node_manager.get_local_nodes( tid ).end();
           ++i )
     {
-      Node* node = i->get_node();
+      StructuralPlasticityNode* node = static_cast< StructuralPlasticityNode* >( i->get_node() );
```

Comment:
Inside this `for-loop`. The `local_nodes` object might contain `Neurons`, `Devices` and `correlomatrix_detector`. Depending on the Simulation script, you have to be careful if the node object is really of type `StructuralPlasticityNode`. Thus, when you do the conversion, you will get a weird object that has undefined behavior. In my testing case, the `for-loop` in the `update_synaptic_elements` was entered despite that it has the size of `zero`.